librz/core/cbin.c

```diff
@@ -1563,7 +1563,7 @@ RZ_API bool rz_core_bin_apply_symbols(RzCore *core, RzBinFile *binfile, bool va)
 				}
 
 				fi = rz_flag_set(core->flags, fnp, addr, symbol->size);
-				if (fi) {
+				if (fi && !RZ_STR_ISEMPTY(n)) {
```

Comment:
According to the code `n` is `const char *n = sn.demname ? sn.demname : symbol->name;`.
`n` to be `NULL` must have `symbol->name` as `NULL`.

This is weird, can you share the binary that does this?